packages/sdk/schema/src/common/message.ts

```diff
@@ -122,6 +123,42 @@ export namespace ContentBlock {
   }).pipe(Schema.mutable);
   export interface ToolResult extends Schema.Schema.Type<typeof ToolResult> {}
 
+  /**
+   * GPT Summary
+   */
+  export const Summary = Schema.TaggedStruct('summary', {
+    mimeType: Schema.optional(Schema.String),
+    message: Schema.String,
+    usage: Schema.optional(
+      Schema.Struct({
+        model: Schema.optional(Schema.String),
+        input: Schema.optional(Schema.Number),
+        output: Schema.optional(Schema.Number),
+        total: Schema.optional(Schema.Number),
+      }),
+    ),
+    duration: Schema.optional(Schema.Number).annotations({
+      description: 'Duration in ms.',
+    }),
+    toolCalls: Schema.optional(Schema.Number),
```

Comment:
could be inferred from the mesage history, what's the value in duplicating it?

Reply:
i think it's fine to roll this up now to make processing simpler given that we have this new message type.